### PR TITLE
feat(ui): show discovery run status on homepage project cards

### LINKED ISSUE #256: Show discovery run status on homepage project list (running with elapsed time, completed with timestamp)
## Summary

On the homepage project list, show the discovery run status for each project. Today each project card shows only a project-status badge and a plain "Last run" date. We want to show the run state clearly:

- **Running** — show that a run is in progress, with elapsed time (for example: `Running — 4m 12s`).
- **Completed** — show the completion timestamp (for example: `Completed 2026-05-29 14:30`).
- **Failed** — show that the last run failed.

## Current behavior

File: `ui/dashboard/src/app/page.tsx`

- Each project is rendered as a card.
- The badge shows `project.status` (the project lifecycle state: `active` / other), **not** the discovery run status.
- The card shows `project.last_run_at` as a date only (lines 79–83):

```tsx
{project.last_run_at && (
  <Text size="xs" c="dimmed" mt="sm">
    Last run: {new Date(project.last_run_at).toLocaleDateString()}
  </Text>
)}
```

## What we want

- Add a clear discovery run status on each project card.
- **Running:** show elapsed time. It should keep counting up while the page is open.
- **Completed:** show the completion timestamp (date + time, not only the date).
- **Failed:** show a failed state.

## Data already available

- `Project` (in `ui/dashboard/src/lib/api.ts`) already has:
  - `last_run_status: string` — **not shown on the card today.**
  - `last_run_at: string | null`
- `DiscoveryRunStatus` (`GET /api/v1/runs/{id}`) has: `status` (`pending` / `running` / `completed` / `failed`), `started_at`, `completed_at`, `progress` (0–100), `phase`.

## Open question for the implementer

The homepage uses `api.listProjects()` only. To show **elapsed time** (now − start time) and the **completion timestamp**, please confirm whether the project-list payload includes the latest run's `started_at` / `completed_at`. If not, we need to either extend the list endpoint to include those fields, or poll `GET /api/v1/runs/{id}` for each project that is currently running. Elapsed time needs the run start time to be present in the list payload.

## Scope

Keep this "simple" for now — a status label plus elapsed time / completion time on the card. A full progress bar or per-phase detail on the homepage is out of scope (that already lives on the discovery run detail page).

### PR DESCRIPTION
**Pull request by abacigil**
## What

The homepage project list now shows each project's **latest discovery-run status** instead of just a plain "Last run" date:

- **Running / pending** — a *Running* badge plus a **live elapsed timer** that counts up every second while the page is open (e.g. `Running — 4m 12s`).
- **Completed** — a *Completed* badge plus the **completion timestamp** (date + time, e.g. `2026-05-29 14:30`).
- **Failed / cancelled** — a clear terminal-state badge.
- **Never run** — nothing (preserves the prior hide-when-empty behaviour).

The lifecycle status badge (`project.status`) in the card header is unchanged — this adds the *run* status below it.

## Why

Per #256, the card only showed `project.status` (the lifecycle state) and a date-only "Last run". The issue's open question was whether the list payload carries the run's `started_at` / `completed_at`. It does **not** — and it turns out the `last_run_status` / `last_run_at` fields the issue references **were never populated server-side**. So this resolves the open question by extending the read path to provide the run summary (the option the issue leaned toward), rather than client-side per-project polling (which would be N+1).

## How

**Server (API)**
- `RunRepository.LatestByProjects` — one **batched aggregation** (`$match $in → $sort started_at desc → $group $first`) returns the latest run per project ID. One Mongo round-trip for a whole page, no N+1. Backed by the existing `{project_id, started_at desc}` index.
- The project `List` and `Get` handlers enrich each project via a new optional `WithRunSummaries(...)` dependency. Enrichment is **best-effort**: with no provider wired, or on a lookup error, the projects are returned unchanged (a run-summary outage must not break the list) — it just logs a warning.
- `models.Project` now exposes `last_run_status`, `last_run_at` (run start, for the elapsed timer), and the new `last_run_completed_at` as **read-only projections** (`bson:"-"`) — derived at read time, never stored on the project document. (The two pre-existing flat fields carried `bson` tags but were never written by any code; they're now meaningful and can't be accidentally persisted.)

**Dashboard**
- New focused, testable `ProjectRunStatus` component (the ticking elapsed timer mirrors the existing `ValidationJobProgressCard` pattern).
- `lib/api.ts`: added `last_run_completed_at` and documented the derived run-summary fields.

## Testing
- **Go unit tests** (`projects_test.go`): List/Get enrichment for running / completed / failed states, never-run projects (no fields), best-effort behaviour on provider error (still 200), and the no-provider path.
- **Go integration tests** (real Mongo testcontainer, `run_repo_integration_test.go`): `LatestByProjects` picks the most-recent run per project, ignores unrequested projects, handles empty input and no-matches. ✅ ran green locally.
- **Jest** (`ProjectRunStatus.test.tsx`): elapsed/timestamp formatters, the live ticker counting up under fake timers, and each status branch.
- Local checks all green: `services/api` `go build` + `go test ./...` + `golangci-lint run` (0 issues), agent + api binaries build, dashboard `npm run build` + `npm test` (303 passed) + `npm run lint` (0 errors), `gofmt` clean.

## Docs
- `docs/reference/api.md` — documents the run-summary fields on the project list/get responses.
- `CHANGELOG.md` — entry under `[Unreleased]`.

Closes #256

— Co-coded with Jale 🤖

### CODE REVIEW

File: services/api/database/run_repo.go [hunk 1]
```diff
@@ -77,6 +77,51 @@ func (r *RunRepository) GetLatestByProject(ctx context.Context, projectID string
...
+// returns an empty map without touching Mongo.
+func (r *RunRepository) LatestByProjects(ctx context.Context, projectIDs []string) (map[string]*models.DiscoveryRun, error) {
+	out := make(map[string]*models.DiscoveryRun, len(projectIDs))
+	if len(projectIDs) == 0 {
+		return out, nil
+	}
+
+	pipeline := mongo.Pipeline{
+		{{Key: "$match", Value: bson.M{"project_id": bson.M{"$in": projectIDs}}}},
+		{{Key: "$sort", Value: bson.D{{Key: "started_at", Value: -1}}}},
```
[the remark this answers is not in the document]
Comment (abacigil) [reply services/api/database/run_repo.go]: Good catch — fixed in b66a1a4. The aggregation now sorts by `{project_id: 1, started_at: -1}`, which exactly matches the existing compound index on `discovery_runs`, so Mongo streams the sort off the index instead of doing a blocking in-memory sort. `$first` within each `project_id` group still yields the latest run; verified unchanged by the `LatestByProjects` integration tests. (Codex flagged the same thing.)

**Comment by abacigil** [comment]
## Final report

**What** — Homepage project cards now show each project's latest discovery-run state instead of a date-only "Last run": a **Running** badge with a live elapsed timer that ticks up every second (`Running — 4m 12s`), **Completed** with the completion timestamp (date + time, `2026-05-29 14:30`), or **Failed** / **Cancelled**. Never-run projects show nothing, matching the prior behaviour.

**How** — The card previously read `project.last_run_status` / `last_run_at`, but those fields were never populated server-side, so they couldn't drive anything. I made the read path supply the data (the option #256 leaned toward, over client-side per-project polling):
- `RunRepository.LatestByProjects` runs a single batched aggregation (`$match $in → $sort {project_id,started_at desc} → $group $first`) to get the latest run per project — one Mongo round-trip per page, no N+1, and the sort key matches the existing `{project_id, started_at}` compound index so there's no blocking in-memory sort.
- The project `List` / `Get` handlers enrich each project (`WithRunSummaries`) with `last_run_status`, `last_run_at` (run start, for elapsed), and the new `last_run_completed_at`. These are read-only projections (`bson:"-"`) — derived at read time, never stored on the project doc. Enrichment is best-effort: a lookup failure logs a warning and returns the list unchanged.
- Dashboard: a focused, testable `ProjectRunStatus` component (the ticker mirrors the existing `ValidationJobProgressCard` pattern); `lib/api.ts` gains `last_run_completed_at`.

**Why** — Resolves #256: the card showed only the project *lifecycle* status, not the *discovery-run* status, and the open question about whether the list payload carries `started_at` / `completed_at` (it did not — and the referenced fields were dead).

**Verification**
- Go unit tests (`projects_test.go`): enrichment for running / completed / failed, never-run (no fields), best-effort on provider error (still 200), and the no-provider path.
- Go integration tests against a real Mongo testcontainer (`run_repo_integration_test.go`): latest-run-per-project selection, unrequested projects excluded, empty input, no-matches — ran green locally.
- Jest (`ProjectRunStatus.test.tsx`, 15 cases): formatters, the live ticker counting up under fake timers, every status branch, and missing-timestamp edge cases.
- Local + CI: `go build` (api + agent), `go test ./...`, `golangci-lint` (0 issues), `gofmt` clean, dashboard `build` + `test` (303 passed) + `lint` (0 errors). **CI green on the latest commit.**

**Review rounds**
- **Codex** — 4 rounds. Round 2 raised one valid P2: the aggregation sorted by `started_at` alone, which doesn't match the compound index and would force a blocking sort (risking the aggregation sort-memory cap on long run histories). Fixed in `b66a1a4` by sorting `{project_id, started_at}`. Rounds 3–4 clean.
- **Copilot** — flagged the same index/sort issue on the first commit (already fixed; replied + resolved). Re-reviewed the latest commit → no new comments. All threads resolved.

**Follow-ups / limitations** — The homepage fetches projects once on load, so a card's status reflects the moment of fetch (the elapsed timer ticks live, but a Running→Completed transition needs a reload). Live auto-transition on the list was intentionally left out of scope per #256 ("keep this simple"); the run detail page already polls for that.

— Co-coded with Jale 🤖
